spyder_kernels/customize/spydercustomize.py

```diff
@@ -303,16 +285,34 @@ def _patched_preparation_data(name):
 #==============================================================================
 # Pdb adjustments
 #==============================================================================
-class SpyderPdb(pdb.Pdb):
+class SpyderPdb(pdb.Pdb, object):  # Inherits `object` to call super() in PY2
 
     send_initial_notification = True
     starting = True
+    def __init__(self, *args, **kwargs):
+        """Init Pdb."""
+        self.continue_if_has_breakpoints = True
+        super(SpyderPdb, self).__init__(*args, **kwargs)
 
     # --- Methods overriden by us
     def preloop(self):
         """Ask Spyder for breakpoints before the first prompt is created."""
-        if self.starting:
-            get_ipython().kernel._ask_spyder_for_breakpoints()
+        try:
+            _frontend_request(blocking=False).set_debug(True)
+            if self.starting:
+                breakpoints = _frontend_request().get_breakpoints()
+                self.set_spyder_breakpoints(breakpoints)
+        except (CommError, TimeoutError):
+            logger.debug("Could not send preloop to frontend.")
```

Comment:
I think this should be

    Could not get breakpoint from the frontend

because that's what's really the problem here.